Mage/src/main/java/mage/players/ManaPool.java

```diff
@@ -34,6 +35,7 @@
     private final List<ManaPoolItem> poolBookmark = new ArrayList<>(); // mana pool bookmark for rollback purposes
 
     private final Set<ManaType> doNotEmptyManaTypes = new HashSet<>();
+    private boolean lastPaymentWasSnow = false;
```

Comment:
Miss from copy constructor: `public ManaPool(final ManaPool pool)`

Reply:
I don't think we need it there, it's just being used to track data after `pay` is used

Reply:
It's a standard practice to copy all inner data on copy (hello rollback all around). If you have dynamic/temporary field then add comment about it (what it doesn't need to copy).

Reply:
alright, sounds reasonable

Reply:
BTW mana payment uses rollback too:

![shot_210118_041302](https://user-images.githubusercontent.com/8344157/104860165-842aa400-5943-11eb-8170-87bbdd74c196.png)